rfcs/0000-new-cloud-fields.md

```diff
@@ -0,0 +1,224 @@
+# 0000: New cloud (or related) fields
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **TBD** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+This RFC is to add several new cloud or cloud related fields to ECS.  As more cloud integrations/data sources are developed there are a variety of fields that don't have an ECS normalized field to put them. This would add the ability to normalize data related to cloud instance images, network/VPC information, S3/Object storage data....
+
+<!--
+Stage 1: If the changes include field additions or modifications, please create a folder titled as the RFC number under rfcs/text/. This will be where proposed schema changes as standalone YAML files or extended example mappings and larger source documents will go as the RFC is iterated upon.
+-->
+
+<!--
+Stage X: Provide a brief explanation of why the proposal is being marked as abandoned. This is useful context for anyone revisiting this proposal or considering similar changes later on.
+-->
+
+## Fields
+
+Initial list of proposed fields:
+
+* `cloud.edge_location`: The edge location, usually an airport IATA code (DFW), of a cloud service. Very common for DNS, CDN, or other cloud services
+* `cloud.instance.image.id`: ID of the image used to create the virtual instance/machine
+* `cloud.instance.image.name`: Name of the image used to create teh virtual instance/machine
+* `cloud.vpc.name`: Name of the virtual network
+* `cloud.vpc.id`: ID of the virtual network
+* `cloud.subnet.name`: Name of the Subnet within the VPC/Virtual network
+* `cloud.subnet.id`: ID of the Subnet within the VPC/Virtual network
+* `cloud.instance.lifecycle`: Type of instance *normal* vs *spot*, see https://github.com/elastic/ecs/issues/323 for more discussion
+* `cloud.object_store.name`: Name of S3/Compatible storage
```

Comment:
I'm unsure about placing `object_store.*` fields underneath `cloud.*`. There are object store solutions like [MinIO](https://min.io/) and [ceph](https://docs.ceph.com/en/quincy/) which wouldn't automatically fit underneath `cloud.*`

Not a blocker for merging the proposal as-is for stage 0, but a topic to discuss and unpack more in a later stage.